test/python/circuit/library/test_evolution_gate.py

```diff
@@ -319,30 +319,31 @@ def test_dag_conversion(self):
     @data("chain", "fountain")
     def test_cnot_chain_options(self, option):
         """Test selecting different kinds of CNOT chains."""
-
-        op = Z ^ Z ^ Z
-        synthesis = LieTrotter(reps=1, cx_structure=option)
-        evo = PauliEvolutionGate(op, synthesis=synthesis)
-
-        expected = QuantumCircuit(3)
-        if option == "chain":
-            expected.cx(2, 1)
-            expected.cx(1, 0)
-        else:
-            expected.cx(1, 0)
-            expected.cx(2, 0)
-
-        expected.rz(2, 0)
-
-        if option == "chain":
-            expected.cx(1, 0)
-            expected.cx(2, 1)
-        else:
-            expected.cx(2, 0)
-            expected.cx(1, 0)
-
-        self.assertEqual(expected, evo.definition)
-        self.assertSuzukiTrotterIsCorrect(evo)
+        for use_sparse_observable in [True, False]:
+            with self.subTest(use_sparse_observable=use_sparse_observable):
+                op = SparseObservable("ZZZ") if use_sparse_observable else SparsePauliOp(["ZZZ"])
```

Comment:
This kind of thing might be neater to do using ddt parametrisation rather than the inner loop.